inference-engine/src/mkldnn_plugin/mkldnn_extension.cpp

```diff
@@ -0,0 +1,204 @@
+//        NGRAPH_OP(PriorBox, ngraph::op::v0)
+//        NGRAPH_OP(PriorBoxClustered, ngraph::op::v0)
+//        NGRAPH_OP(Proposal, ngraph::op::v0)
+//        NGRAPH_OP(Range, ngraph::op::v0)
+        NGRAPH_OP(Relu, ngraph::op::v0)
+        NGRAPH_OP(ReduceMax, ngraph::op::v1)
+        NGRAPH_OP(ReduceLogicalAnd, ngraph::op::v1)
+        NGRAPH_OP(ReduceLogicalOr, ngraph::op::v1)
+        NGRAPH_OP(ReduceMean, ngraph::op::v1)
+        NGRAPH_OP(ReduceMin, ngraph::op::v1)
+//        NGRAPH_OP(ReduceProd, ngraph::op::v1)
+        NGRAPH_OP(ReduceSum, ngraph::op::v1)
+//        NGRAPH_OP(RegionYolo, ngraph::op::v0)
+        NGRAPH_OP(Reshape, ngraph::op::v1)
+//        NGRAPH_OP(Result, ngraph::op::v0)
+//        NGRAPH_OP(Reverse, ngraph::op::v1)
+//        NGRAPH_OP(ReverseSequence, ngraph::op::v0)
+//        NGRAPH_OP(RNNCell, ngraph::op::v0)
+        NGRAPH_OP(Select, ngraph::op::v1)
+//        NGRAPH_OP(Selu, ngraph::op::v0)
+        NGRAPH_OP(ShapeOf, ngraph::op::v0)
+        NGRAPH_OP(ShuffleChannels, ngraph::op::v0)
+//        NGRAPH_OP(Sign, ngraph::op::v0)
+//        NGRAPH_OP(Sigmoid, ngraph::op::v0)
+//        NGRAPH_OP(Sin, ngraph::op::v0)
+//        NGRAPH_OP(Sinh, ngraph::op::v0)
+//        NGRAPH_OP(Softmax, ngraph::op::v1)
+//        NGRAPH_OP(Sqrt, ngraph::op::v0)
+//        NGRAPH_OP(SpaceToDepth, ngraph::op::v0)
+//        NGRAPH_OP(Split, ngraph::op::v1)
+//        NGRAPH_OP(SquaredDifference, ngraph::op::v0)
+        NGRAPH_OP(Squeeze, ngraph::op::v0)
+//        NGRAPH_OP(StridedSlice, ngraph::op::v1)
+        NGRAPH_OP(Subtract, ngraph::op::v1)
+//        NGRAPH_OP(Tan, ngraph::op::v0)
+//        NGRAPH_OP(Tanh, ngraph::op::v0)
+//        NGRAPH_OP(Tile, ngraph::op::v0)
+//        NGRAPH_OP(TopK, ngraph::op::v1)
+//        NGRAPH_OP(Transpose, ngraph::op::v1)
+        NGRAPH_OP(Unsqueeze, ngraph::op::v0)
+//        NGRAPH_OP(VariadicSplit, ngraph::op::v1)
+//        NGRAPH_OP(Xor, ngraph::op::v0)
+        NGRAPH_OP(MVN, ngraph::op::v0)
+        NGRAPH_OP(MVN, ngraph::op::v6)
+        NGRAPH_OP(Select, ngraph::op::v1)
+        NGRAPH_OP(ConvolutionBackpropData, ngraph::op::v1)
+#undef NGRAPH_OP
+
+        return opset;
+    };
+
+    static std::map<std::string, ngraph::OpSet> opsets = {
+        { "cpu_plugin_opset", cpu_plugin_opset() },
+        { "type_relaxed_opset", type_relaxed_opset() }
+    };
+
+    return opsets;
+}
+
+std::vector<std::string> MKLDNNExtension::getImplTypes(const std::shared_ptr<ngraph::Node>&) {
+    return {};
+}
+
+InferenceEngine::ILayerImpl::Ptr MKLDNNExtension::getImplementation(const std::shared_ptr<ngraph::Node>& node, const std::string& implType) {
+    return nullptr;
+}
+
+}  // namespace MKLDNNPlugin
+
+// Generate exported function
+IE_DEFINE_EXTENSION_CREATE_FUNCTION(MKLDNNPlugin::MKLDNNExtension)
+
+INFERENCE_EXTENSION_API(InferenceEngine::StatusCode)
+InferenceEngine::CreateExtension(InferenceEngine::IExtension*& ext, InferenceEngine::ResponseDesc* resp) noexcept {
```

Comment:
If I understand right, it some internal extensions. I don't think that we need to create a new entry point to the plugin, users cannot load the plugin as a shared library with extensions.

Reply:
Unfortunately, the lifetime of the required extensions that are used for deserialization and the lifetime of the plugin are different. The plugin can be unloaded at any time. Therefore, the plugin is also added as an extension library if it provides the required extension interface.